fdbclient/include/fdbclient/CommitTransaction.h

```diff
@@ -157,11 +157,20 @@ struct MutationRef {
 		    AES_256_IV_LENGTH,
 		    getEncryptAuthTokenMode(EncryptAuthTokenMode::ENCRYPT_HEADER_AUTH_TOKEN_MODE_SINGLE),
 		    usageType);
-		BlobCipherEncryptHeader* header = new (arena) BlobCipherEncryptHeader;
-		StringRef headerRef(reinterpret_cast<const uint8_t*>(header), sizeof(BlobCipherEncryptHeader));
-		StringRef payload =
-		    cipher.encrypt(static_cast<const uint8_t*>(bw.getData()), bw.getLength(), header, arena)->toStringRef();
-		return MutationRef(Encrypted, headerRef, payload);
+		if (CLIENT_KNOBS->ENABLE_CONFIGURABLE_ENCRYPTION) {
+			BlobCipherEncryptHeaderRef headerRef;
+			StringRef payload =
+			    cipher.encrypt(static_cast<const uint8_t*>(bw.getData()), bw.getLength(), &headerRef, arena);
+			Standalone<StringRef> serializedHeader = BlobCipherEncryptHeaderRef::toStringRef(headerRef);
+			arena.dependsOn(serializedHeader.arena());
```

Comment:
nit: Add a version of `BlobCipherEncryptHeaderRef::toStringRef` that would accept an arena?

Reply:
sure, we can do it.